digitalocean/resource_digitalocean_uptime_alert.go

```diff
@@ -0,0 +1,221 @@
+package digitalocean
+
+import (
+	"context"
+	"log"
+
+	"github.com/digitalocean/godo"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceDigitalOceanUptimeAlert() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceDigitalOceanUptimeAlertCreate,
+		ReadContext:   resourceDigitalOceanUptimeAlertRead,
+		UpdateContext: resourceDigitalOceanUptimeAlertUpdate,
+		DeleteContext: resourceDigitalOceanUptimeAlertDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "A human-friendly display name for the alert.",
+				Required:    true,
+			},
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"check_id": {
+				Type:        schema.TypeString,
+				Description: "A unique identifier for a check.",
+				Required:    true,
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Description: "The type of health check to perform. Enum: 'latency' 'down' 'down_global' 'ssl_expiry'",
```

Comment:
For some of these attributes  that are enums, it could make sense to do some input validation. For example:

```suggestion
				Description: "The type of health check to perform. Enum: 'latency' 'down' 'down_global' 'ssl_expiry'",
				ValidateFunc: validation.StringInSlice([]string{
					"latency",
					"down",
					"down_global",
					"ssl_expiry",
				}, false),
```

Reply:
thank you!